### ISSUE TITLE
Performance problem: Deleting of processes

### ISSUE THREAD

**Issue by andre-hohmann** [kitodo/kitodo-production#4782]
### Problem
It takes several seconds until a process is deleted. This does not strike, if only few processes are deleted. If several processes have to be deleted, it is quite uncomfortable and time consuming. 

### Solution
The deletion of processes should be quicker. 

Relevant issues: 
- #3728 
- #4589

**Comment by andre-hohmann** [comment]
@Kathrin-Huber: 

Yes, this is still a problem. It takes several seconds to delete a process.

**Comment by BartChris** [comment]
It seems like deletion of processes got even slower on current master. (Maybe Hibernate Search related?) @matthias-ronge : Is the system now doing more than before?

**Comment by BartChris** [comment]
The new deletion of process seems to do way more then the old one:

Before Hibernate Search:

https://github.com/kitodo/kitodo-production/blob/fdd9b719e3297ff51758de6b5dea65a14bbc61d9/Kitodo/src/main/java/org/kitodo/production/services/data/ProcessService.java#L2351-L2379

After Hibernate Search:

https://github.com/kitodo/kitodo-production/blob/0ff74d70d044eb6eec5cb0f543532d4fd0d3e6d7/Kitodo/src/main/java/org/kitodo/production/services/data/ProcessService.java#L1814-L1859

cc @henning-gerhardt

**Comment by henning-gerhardt** [comment]
@BartChris : it's look more complex and may be needed as long as we delete it on our way instead of defining deleting data in a better way through annotations so it can be done by the used framework and / or by proper SQL statements so it can be deleted by the database itself if the process is deleted. There are many ways to delete something and the current way is one of them - may be not the best.

**Comment by BartChris** [comment]
What takes long to delete right now are processes based on a catalog import, as those have a lot of properties, which are unfortunately duplicated right now (https://github.com/kitodo/kitodo-production/issues/6573). Deleting an empty process (one without images) based on a catalogue import takes 5 seconds, so deleting 20 of those would take 100 seconds, which is quite slow. On 3.8 deleting a process based on the same catalogue entry takes less than 1 second.

**Comment by henning-gerhardt** [comment]
As I must now delete some processes on the main branch I can reproduce this slow deleting too. Already 2 or 3 minutes are gone but deleting one process (part of a hierarchy) is not deleted :-(

**Comment by BartChris** [comment]
> [@BartChris](https://github.com/BartChris) : I tried your suggestion but I did not see a major difference in the execution time of deleting a process.

The general problem here is an absolute explosion of SQL queries (automatically triggered by Hibernate while doing related operaitions during the deletion) and i suppose the SQL queries done by the "@form" annotation are in large databases like yours insignificant compared to the large number of queries done during the actual deletion.

**Comment by henning-gerhardt** [comment]
@solth : can this issue be added to the 3.9.x milestone for the next release? Deleting processes should not take so much time after Hibernate-Search was integrated.

**Comment by BartChris** [comment]
@henning-gerhardt Can you maybe already try my PR?

https://github.com/kitodo/kitodo-production/pull/6601

i have to do some more tests, if anything works as expected, but in my first tests functionality wise it worked. Will comment on it there, too.
